activerecord/lib/active_record/persistence.rb

```diff
@@ -378,6 +378,10 @@ def _insert_record(values) # :nodoc:
       def _update_record(values, constraints) # :nodoc:
         constraints = _substitute_values(constraints).map { |attr, bind| attr.eq(bind) }
 
+        if default_scopes.any? && !default_scoped(all_queries: true).where_clause.empty?
+          constraints << default_scoped(all_queries: true).where_clause.ast
+        end
```

Comment:
I'm not sure this is the best way to do this. I didn't want to merge `where_values_hash` in the methods that call `_update_record` because then that's updating a bunch of different places. However once we're at this part of the code `constraints` is already an array of Equality nodes, so we need to use the where clause ast to also have an Equality node to merge into constraints.